services/explorer/src/hooks/useExplorerNetwork.tsx

```diff
@@ -41,34 +34,34 @@ const initialState: Context = {
 const ExplorerNetworkContext = createContext<Context>(initialState);
 
 const useNetworkSelection = () => {
-  const networks = Object.values(Network);
+  const networks: string[] = Object.values(NetworkName);
   const hostNetwork = getDomainName();
-  const [currentNetwork, setCurrentNetwork] = useState<string>(
-    hostNetwork || Network.MAINNET
+  const [currentNetwork, setCurrentNetwork] = useState<NetworkName>(
+    hostNetwork || NetworkName.MAINNET
   );
 
   const [graphQLClient, setGraphQLClient] = useState<Client>(
-    createClient(recreateGraphQLConnection(currentNetwork))
+    createClient(getNetwork(currentNetwork).graphqlUrl)

```

Comment:
```suggestion
    const selectedNetwork = getNetwork(currentNetwork);
    createClient(selectedNetwork.graphqlUrl)
 etc...
```
So we don't getNetwork(),getNetwork(),getNetwork() ..